src/jimgw/core/single_event/likelihood.py

```
@@ -913,7 +969,773 @@ def _step(carry, _):
+    def _find_leaf_prior(self, prior: Prior, param_name: str) -> Optional[Prior]:
+        """Recursively search *prior* for the bounded leaf that owns *param_name*.
+
+        Returns the first component that has ``xmin``/``xmax`` attributes and
+        lists *param_name* in its ``parameter_names``, or ``None`` if not found.
+        """
+        if param_name not in prior.parameter_names:
+            return None
+        if hasattr(prior, "xmin"):
+            return prior
+        if hasattr(prior, "base_prior"):
+            components = getattr(prior, "base_prior")
```

Ruff (0.15.15)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)